annotator-scanner/src/main/java/edu/ucr/cs/riple/scanner/AnnotatorScanner.java

```diff
@@ -165,4 +222,23 @@ private void serializeSymIfField(Symbol symbol, VisitorState state) {
           .serializeFieldGraphNode(new TrackerNode(context.getConfig(), symbol, state.getPath()));
     }
   }
+
+  /**
+   * Serializes the impacted region for a functional interface. This method locates the overriden
+   * method from the passed functional interface tree and serializes the impacted region for that
+   * method.
+   *
+   * @param tree Given tree.
+   * @param state Visitor State.
+   */
+  private static void serializeImpactedRegionForFunctionalInterface(
+      Config config, ExpressionTree tree, VisitorState state) {
+    Symbol.MethodSymbol methodSym = SymbolUtil.getFunctionalInterfaceMethod(tree, state.getTypes());
+    if (methodSym == null) {
+      return;
+    }
```

Comment:
I think the method symbol should always be non-null here.  Should we throw an exception if we have `null` rather than just returning?

Reply:
@msridhar I am not sure about throwing an exception in Scanner. Throwing an exception here will terminate the serialization, but Annotator cannot distinguish if the serialized output is complete or incomplete (terminated by an exception) and will work on whatever serialized data which leads to bad outcomes by Annotator. The only way I can think of is to pars standard error while building the target and monitor it to see if any exception is thrown by Scanner. Please let me know if you think I should implement this interaction between Annotator and Scanner.

Reply:
ok, it would be nice if we could at least print or log something, but I'm ok with leaving this as just a return I guess

Reply:
Sure, makes sense. f5792cee75e1c2b804341623668bb8e66f3a77e3